### DISCUSSION TITLE
Can be used with FlowFuse Cloud ?

### DISCUSSION THREAD

**Discussion by muratcanberber** [TotallyInformation/node-red-contrib-uibuilder#230] [General]
I am trying discover FlowFuse and wondering  if UIBuilder can be use in the FlowFuse environment ?

**Comment by TotallyInformation** [comment]
Hi, thanks for reaching out. Unfortunately, FlowFuse decided to implement an envionment that does not contain a real filing system. They have some built-in workaround that lets core Node-RED and their nodes to work but UIBUILDER currently relies so heavily on the filing system that I don't believe it will work. 

Not sure if that holds true for a self-hosted version though.

I did try to work with them to see if we could fix this but unfortunately, they weren't really interested in working in partnership. Even though they were interested in UIBUILDER possibly being the next Dashboard (this was well before Dashboard v2 was announced), they would only offer a "free month's membership". So that didn't go anywhere.

There are tools that could be used to create a virtual FS and indeed, I believe they use one internally but without additional support from them to do the integration, while I may be able to get something working, I doubt it would integrate.

In the meantime, what I am doing is slowly moving all of the filing system functions to a central library so that any future work on virtual FS's would become much easier. It will also let me remove a dependency on the fs-extra 3rd-party library.

**Comment by TotallyInformation** [comment]
I should have added an update to this thread some time ago.

Turns out that FlowFuse could indeed enable the ability for UIBUILDER and other nodes that need filing system access to work. That was implemented and so UIBUILDER should indeed work just fine on FlowFuse.
